config.yaml

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# This file defines charm config options, and populates the Configure tab on Charmhub.
+# If your charm does not require configuration options, delete this file entirely.
+#
+# See https://juju.is/docs/config for guidance.
+
+options:
+  external-hostname:
+    description: |
+        The DNS listing used for external connections. 
+        Will default to the name of the deployed application.
+    type: string
+    default: superset-k8s
+  tls-secret-name:
+    description: |
+        Name of the k8s secret which contains the TLS certificate to be used by ingress.
+    default: superset-tls
+    type: string
+  superset-secret-key:
+    description: |
+        Your App secret key will be used for securely signing the session cookie
+        and encrypting sensitive information on the database.
+    type: string
+  admin-password:
+    description: |
+        The secret for the admin account created on app initialisation. Default is 'admin'.
+    default: admin
+    type: string
+  charm-function:
+    description: |
+        The type of superset function you want the charm to fulfill.
+        Allowed options are: 'app-gunicorn', 'app', 'worker', 'beat'
```

Comment:
Just to double check, I remember reading there were three functions instead of four?

Reply:
I see four functions later on in the PR. Do you mean outside of the charm code?

Reply:
@kelkawi-a I think you read about `app-gunicorn`, `worker` and `beat` in the spec doc. I did not mention `app` there as this is used instead of the gunicorn web server for local development but is not suitable for production. I only discussed the production setup in the doc. I'll add a note to the spec.